core/trino-main/src/main/java/io/trino/operator/scalar/JsonExistsFunction.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+import io.trino.annotation.UsedByGeneratedCode;
+import io.trino.metadata.BoundSignature;
+import io.trino.metadata.FunctionMetadata;
+import io.trino.metadata.FunctionNullability;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.Signature;
+import io.trino.metadata.SqlScalarFunction;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.TypeSignature;
+import io.trino.type.JsonPath2016Type;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.trino.metadata.FunctionKind.SCALAR;
+import static io.trino.metadata.Signature.typeVariable;
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.BOXED_NULLABLE;
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+import static io.trino.spi.type.StandardTypes.BOOLEAN;
+import static io.trino.spi.type.StandardTypes.JSON_2016;
+import static io.trino.spi.type.StandardTypes.TINYINT;
+import static io.trino.util.Reflection.methodHandle;
+import static java.util.Objects.requireNonNull;
+
+public class JsonExistsFunction
+        extends SqlScalarFunction
+{
+    public static final String FUNCTION_NAME = "$json_exists";
```

Comment:
nit: I would slightly prefer `JSON_EXISTS_FUNCTION_NAME` and then use elsewhere with static import

Reply:
applied to all